File: logisim/app/src/main/java/logisim/std/memory/DFlipFlop.java
/* Copyright (c) 2010, Carl Burch. License information is located in the
 * logisim.Main source code and at www.cburch.com/logisim/. */

package logisim.std.memory;


import logisim.data.WireValue.WireValue;

public class DFlipFlop extends AbstractFlipFlop {
	public DFlipFlop() {
		super("D Flip-Flop", "dFlipFlop.gif", Strings.getter("dFlipFlopComponent"), 1, true);
	}

	@Override
	protected String getInputName(int index) {
		return "D";
	}

	@Override
	protected WireValue computeValue(WireValue[] inputs, WireValue curValue) {
		return inputs[0];
	}
}

File: LogisimPlus/LogisimPlus/logisim/std/memory/DFlipFlop.cs
// ====================================================================================================
// Produced by the Free Edition of Java to C# Converter.
// To produce customized conversions, purchase a Premium Edition license:
// https://www.tangiblesoftwaresolutions.com/product-details/java-to-csharp-converter.html
// ====================================================================================================

/* Copyright (c) 2010, Carl Burch. License information is located in the
 * logisim.Main source code and at www.cburch.com/logisim/. */

namespace logisim.std.memory
{
	using Value = logisim.data.Value;

	public class DFlipFlop : AbstractFlipFlop
	{
		public DFlipFlop() : base("D Flip-Flop", "dFlipFlop.gif", Strings.getter("dFlipFlopComponent"), 1, true)
		{
		}

		protected internal override string getInputName(int index)
		{
			return "D";
		}

		protected internal override Value computeValue(Value[] inputs, Value curValue)
		{
			return inputs[0];
		}
	}

}
